### PR TITLE
Fix issue #38366 salt-ssh support private key's passphrase

### PR DESCRIPTION
**Pull request by pengyao**
### What does this PR do?

Fix issue #38366

### What issues does this PR fix or reference?

Support private key's passphrase for salt-ssh

### Previous Behavior

Not support

### New Behavior

Private key's passphrase could be configured by:
* `--priv-passwd` for salt-ssh cli
* `salt_priv_passwd` for salt master configure file
* `priv_passwd` for salt roster file

### CODE REVIEW

**Review by gtmanfred — CHANGES_REQUESTED** [review]
Can you add docs about this config option to the salt ssh topics page?

**Review by cachedout — APPROVED** [review]

**Comment by gtmanfred** [comment]
Also, adding this to the release notes for Fluorine would be greatly appreciated.

File: doc/man/salt-ssh.1 [hunk 1]
```diff
@@ -177,6 +177,11 @@ Specify the SSH private key file to be used for authentication.
 .UNINDENT
 .INDENT 0.0
 .TP
+.B \-\-priv\-passwd=SSH_PRIV_PASSWD
+Specify the SSH private key file's passphrase.
```
Comment (gtmanfred) [inline doc/man/salt-ssh.1]: Can you remove this change?

We compile these when they are packaged, it does not need to be updated.

**Review by gtmanfred — CHANGES_REQUESTED** [review]

**Review by gtmanfred — APPROVED** [review]

**Comment by rallytime** [comment]
This change is causing many of the ssh tests to fail. @pengyao Can you take a look?

**Comment by pengyao** [comment]
@rallytime It's my fault. I have fixed it.

**Comment by rallytime** [comment]
@pengyao Looks like one more test here needs to be addressed :)

https://jenkins.saltstack.com/job/PR/job/salt-pr-linode-ubuntu14-n/22289/

**Comment by cachedout** [comment]
@pengyao Are you able to resolve that final failing test? We would really like to get this merged if we can. :)

**Comment by pengyao** [comment]
@cachedout @rallytime  Jenkins Error is: 
```
No test reports found for the metric 'JUnit' with the resolved pattern 'artifacts/unittests/*.xml'. Configuration error?.
```
I think the error is not created by this commits

**Comment by rallytime** [comment]
re-run py

**Comment by rallytime** [comment]
@pengyao The following test is still failing:

- unit.config.schemas.test_ssh.RosterEntryConfigTest.test_config

That will need to be fixed before we can merge this in.

https://jenkins.saltstack.com/job/PR/job/salt-pr-linode-ubuntu14-n/23028/

**Comment by rallytime** [comment]
Thanks for fixing that @pengyao - that looks good.

I think an addition to the release notes (for Fluorine) would be good here as well. Can you add something for that? Then we can get this in :)

**Comment by rallytime** [comment]
@pengyao We just need this added to the [Fluorine release notes](https://github.com/saltstack/salt/blob/develop/doc/topics/releases/fluorine.rst) and then we can get this in.

**Comment by rallytime** [comment]
@pengyao Oh! You're right. I thought I double checked but I must have missed it. Thank you!

**Pull request merged**